error: cannot initialize a member subobject of type 'sector_id_t' (aka 'unsigned long') with an rvalue of type 'std::nullptr_t'

Hi,
I compiled the last tag 0.6.0 and build failed on FreeBSD 14.2.

FreeBSD clang version 18.1.6 (https://github.com/llvm/llvm-project.git llvmorg-18.1.6-0-g1118c2e05e67)
Target: x86_64-unknown-freebsd14.2
Thread model: posix
InstalledDir: /usr/bin


```
FAILED: libopenage/CMakeFiles/libopenage.dir/pathfinding/pathfinder.cpp.o 
/usr/bin/c++ -DQT_CORE_LIB -DQT_GUI_LIB -DQT_MULTIMEDIA_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_OPENGL_LIB -DQT_QMLBUILTINS_LIB -DQT_QMLINTEGRATION_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICK_LIB -Dlibopenage_EXPORTS -I/usr/ports/games/openage/work/.build/libopenage/libopenage_autogen/include -I/usr/ports/games/openage/work/.build/libopenage -I/usr/ports/games/openage/work/openage-0.6.0/libopenage -I/usr/local/include/freetype2 -I/usr/local/include/opus -I/usr/ports/games/openage/work/.build/libopenage/coord -isystem /usr/local/include -isystem /usr/local/include/eigen3 -isystem /usr/local/include/qt6/QtCore -isystem /usr/local/include/qt6 -isystem /usr/local/lib/qt6/mkspecs/freebsd-clang -isystem /usr/local/include/qt6/QtQuick -isystem /usr/local/include/qt6/QtGui -isystem /usr/local/include/qt6/QtQml -isystem /usr/local/include/qt6/QtQmlIntegration -isystem /usr/local/include/qt6/QtQmlBuiltins -isystem /usr/local/include/qt6/QtNetwork -isystem /usr/local/include/qt6/QtQmlModels -isystem /usr/local/include/qt6/QtOpenGL -isystem /usr/local/include/qt6/QtMultimedia -O2 -pipe -fstack-protector-strong -fno-strict-aliasing -DNDEBUG -O3 -std=gnu++20 -fPIC  -Wall -Wextra -pedantic -Wno-gnu-statement-expression -pthread -fPIC -MD -MT libopenage/CMakeFiles/libopenage.dir/pathfinding/pathfinder.cpp.o -MF libopenage/CMakeFiles/libopenage.dir/pathfinding/pathfinder.cpp.o.d -o libopenage/CMakeFiles/libopenage.dir/pathfinding/pathfinder.cpp.o -c /usr/ports/games/openage/work/openage-0.6.0/libopenage/pathfinding/pathfinder.cpp
/usr/ports/games/openage/work/openage-0.6.0/libopenage/pathfinding/pathfinder.cpp:473:15: error: cannot initialize a member subobject of type 'sector_id_t' (aka 'unsigned long') with an rvalue of type 'std::nullptr_t'
  473 |         entry_sector{nullptr},
      |                      ^~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.

```

> This seems strange as when I look in the [code ](https://github.com/SFTtech/openage/blob/faec07da10f72f7f7f69b280fe0f6c65960ddffb/libopenage/pathfinding/pathfinder.cpp#L473) entry_sector is initialized with NULL not nullptr

I had right 
```
/usr/ports/games/openage/work/openage-0.6.0/libopenage/pathfinding/pathfinder.cpp:473:15: error: cannot initialize a member subobject of type 'sector_id_t' (aka 'unsigned long') with an rvalue of type 'std::nullptr_t'
  473 |         entry_sector{NULL},
      |                      ^~~~
/usr/include/sys/_null.h:35:14: note: expanded from macro 'NULL'
   35 | #define NULL    nullptr
      |                 ^~~~~~~
1 error generated.
```

--- a/libopenage/pathfinding/pathfinder.cpp
+++ b/libopenage/pathfinding/pathfinder.cpp
@@ -269,11 +269,12 @@ const Pathfinder::portal_star_t Pathfinder::portal_a_star(const PathRequest &req
 		}
 
 		// get the exits of the current node
-		const auto &exits = current_node->get_exits(current_node->entry_sector);
+		ENSURE(current_node->entry_sector != std::nullopt, "Entry sector not set for portal node.");
+		const auto &exits = current_node->get_exits(current_node->entry_sector.value());
 
 		// evaluate all neighbors of the current candidate for further progress
 		for (auto &[exit, distance_cost] : exits) {
-			exit->entry_sector = current_node->portal->get_exit_sector(current_node->entry_sector);
+			exit->entry_sector = current_node->portal->get_exit_sector(current_node->entry_sector.value());
 			bool not_visited = !visited_portals.contains(exit->portal->get_id());
 
 			if (not_visited) {
@@ -289,9 +290,9 @@ const Pathfinder::portal_star_t Pathfinder::portal_a_star(const PathRequest &req
 			if (not_visited or tentative_cost < exit->current_cost) {
 				if (not_visited) {
 					// Get heuristic cost (from exit node to target cell)
-					auto exit_sector = grid->get_sector(exit->portal->get_exit_sector(exit->entry_sector));
+					auto exit_sector = grid->get_sector(exit->portal->get_exit_sector(exit->entry_sector.value()));
 					auto exit_sector_pos = exit_sector->get_position().to_tile(sector_size);
-					auto exit_portal_pos = exit->portal->get_exit_center(exit->entry_sector);
+					auto exit_portal_pos = exit->portal->get_exit_center(exit->entry_sector.value());
 					exit->heuristic_cost = Pathfinder::heuristic_cost(
 						exit_sector_pos + exit_portal_pos,
 						request.target);
@@ -469,7 +470,7 @@ int Pathfinder::distance_cost(const coord::tile_delta &portal1_pos,
 
 PortalNode::PortalNode(const std::shared_ptr<Portal> &portal) :
 	portal{portal},
-	entry_sector{NULL},
+	entry_sector{std::nullopt},
 	future_cost{std::numeric_limits<int>::max()},
 	current_cost{std::numeric_limits<int>::max()},
 	heuristic_cost{std::numeric_limits<int>::max()},

--- a/libopenage/pathfinding/pathfinder.h
+++ b/libopenage/pathfinding/pathfinder.h
@@ -4,6 +4,7 @@
 
 #include <map>
 #include <memory>
+#include <optional>
 #include <unordered_map>
 
 #include "coord/tile.h"
@@ -209,7 +210,7 @@ public:
 	/**
 	 * Sector where the portal is entered.
 	 */
-	sector_id_t entry_sector;
+	std::optional<sector_id_t> entry_sector;
 
 	/**
 	 * Future cost estimation value for this node.

